src/components/outside_click_detector/outside_click_detector.stories.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { EuiText } from '../text';
+import {
+  EuiOutsideClickDetector,
+  EuiOutsideClickDetectorProps,
+} from './outside_click_detector';
+
+const meta: Meta<EuiOutsideClickDetectorProps> = {
+  title: 'Utilities/EuiOutsideClickDetector',
+  component: EuiOutsideClickDetector,
+};
+
+export default meta;
+type Story = StoryObj<EuiOutsideClickDetectorProps>;
+
+export const Playground: Story = {
+  args: {
+    children: (
+      <EuiText>
+        <p>Click anywhere outside of this text to trigger an alert</p>
+      </EuiText>
+    ),
+    onOutsideClick: () => {
+      window.alert('Clicked outside');
+    },
```

Comment:
Sadly this makes it so a Storybook action doesn't automatically fire in the actions pane - any chance we can create one manually and call it here? e.g.

```js
import { action } from '@storybook/addon-actions';
```
```suggestion
    onOutsideClick: (...args) => {
      action('onOutsideClick')(...args);
      window.alert('Clicked outside');
    },
```